pom.xml

```diff
@@ -231,8 +231,43 @@
                     </execution>
                 </executions>
                 <configuration>
-                    <source>16</source>
+                    <source>${java.version}</source>
```

Comment:
This should probably be hard-coded so people who aren't using 16 aren't making PRs thinking their java version is what we're using.